test/unit/services/aws/credentials_service_test.rb

```diff
@@ -7,45 +7,46 @@ class Aws::CredentialsServiceTest < ActiveSupport::TestCase
     File.stubs(:exist?).returns(true)
   end
 
+  test '#call returns IAM credentials when available' do
+    Aws::AssumeRoleWebIdentityCredentials.expects(:new).never
+
+    assert Aws::CredentialsService.call(iam_auth_params), {
+      access_key_id: iam_auth_params[:access_key_id],
+      secret_access_key: iam_auth_params[:secret_access_key]
+    }
+  end
```

Comment:
Would be good to have a test that verifies IAM preference over STS.

Reply:
Agreed! Added in 16f2b72.